repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/StakeholderTitlePreProcessor.java

```diff
@@ -114,12 +115,12 @@ private void processCreateStakeholderTitle(AtlasEntity entity) throws AtlasBaseE
             if (CollectionUtils.isEmpty(domainQualifiedNames)) {
                 throw new AtlasBaseException(BAD_REQUEST, "Please provide attribute " + ATTR_DOMAIN_QUALIFIED_NAMES);
             }
-
-            if (domainQualifiedNames.contains(STAR)) {
+            domainQualifiedNames.replaceAll(s -> s.equals(STAR) ? NEW_STAR : s);
```

Comment:
instead of replace, add another OR condition is next if

Reply:
```
if (domainQualifiedNames.contains(NEW_STAR) || domainQualifiedNames.contains(STAR)) {
                if (domainQualifiedNames.size() > 1) {

                    domainQualifiedNames.clear();
                    domainQualifiedNames.add(NEW_STAR);
                    entity.setAttribute(ATTR_DOMAIN_QUALIFIED_NAMES, domainQualifiedNames);
                }

                String qualifiedName = format(PATTERN_QUALIFIED_NAME_ALL_DOMAINS, getUUID());
                entity.setAttribute(QUALIFIED_NAME, qualifiedName);

            } else {
                entity.setAttribute(QUALIFIED_NAME, format(PATTERN_QUALIFIED_NAME_DOMAIN, getUUID()));
            }

            authorizeDomainAccess(domainQualifiedNames);
```

in this case when `authorizeDomainAccess(domainQualifiedNames);` is called when 
`domainQualifiedNames.size() <= 1`  `STAR` will be authorised instead of the new pattern `NEW_STAR`
Will this work?

Reply:
right, understood, thanks for explanation, this comment can be ignored

Reply:
Then, let us move this inside i condition before following

String qualifiedName = format(PATTERN_QUALIFIED_NAME_ALL_DOMAINS, getUUID());